docs/content/3.middleware/2.guides/11.file-upload.md

```diff
@@ -0,0 +1,200 @@
+  }
+
+  const client = await context.getApiClient("commerce");
+  const formData = new FormData();
+
+  // Handle both multiple files and single file uploads
+  if (Array.isArray(files)) {
+    files.forEach((file) => {
+      formData.append("files", new Blob([file.buffer]), file.originalname);
+    });
+  }
+
+  // Send files to your external service endpoint
+  const response = await client.api.post(
+    "/your-external-service/upload",
+    formData,
+    {
+      headers: {
+        "Content-Type": "multipart/form-data",
+      },
+    }
+  );
+
+  return response.data;
+}
+```
+
+## Usage with SDK
+
+::alert{type="success"}
+For detailed instructions on using file upload with the SDK, visit the [File Upload SDK Guide](https://docs.alokai.com/sdk/advanced/file-upload).
+::
+
+## Custom Configuration
+
+You can customize the file upload behavior by modifying the middleware options:
+
+```typescript
+// middleware.config.js
+module.exports = {
+  integrations: {
+    commerce: {
+      location: "@vue-storefront/commerce-api",
+    },
+  },
+  // Custom file upload configuration
+  fileUpload: {
+    enabled: true,
+    maxFileSize: 10 * 1024 * 1024, // 10MB
+    maxFiles: 10,
+    allowedMimeTypes: ["image/*", "application/pdf", "text/plain"],
+    fieldNames: ["avatar", "documents"], // Specific field names to accept
+  },
+};
+```
+
+::tip
+The middleware automatically handles file size limits, MIME type validation, and file count restrictions based on your configuration.
+::
+
+## Additional Parameters
+
+When uploading files, you can include additional parameters alongside your files. These parameters can be accessed in your middleware implementation:
+
+```typescript
+// api/custom-methods/uploadFile.ts
+export async function uploadFile(
+  context: IntegrationContext
+): Promise<CustomMethodResponse> {
+  const files = context.req.files;
+  const additionalParams = context.req.body; // Access additional parameters
+
+  if (!files) {
+    throw new Error("No files were uploaded");
+  }
+
+  const client = await context.getApiClient("commerce");
+  const formData = new FormData();
+
+  // Add additional parameters to formData
+  Object.entries(additionalParams).forEach(([key, value]) => {
+    formData.append(key, value);
+  });
+
+  // Handle files
+  if (Array.isArray(files)) {
+    files.forEach((file) => {
+      formData.append("files", new Blob([file.buffer]), file.originalname);
+    });
+  }
+
+  // Send files and additional parameters to your external service
+  const response = await client.api.post(
+    "/your-external-service/upload",
+    formData,
+    {
+      headers: {
+        "Content-Type": "multipart/form-data",
+      },
+    }
+  );
+
+  return response.data;
+}
+```
+
+Example of sending files with additional parameters from the client side:
+
+```typescript
+const formData = new FormData();
+formData.append("files", fileInput.files[0]);
+formData.append("userId", "12345");
```

Comment:
@bartoszherba did this get updated? I recall I couldn't pass formData from the frontend via the SDK. Cool if it does work.

Reply:
i removed that